data_steward/cdr_cleaner/cleaning_rules/remove_participant_data_past_deactivation_date.py

```diff
@@ -0,0 +1,164 @@
+    """
+    Sandboxes and drops all EHR data found for deactivated participants after their deactivation date
+
+    :param client: BQ client
+    :param api_project_id: Project containing the RDR Participant Summary API
+    :param project_id: Identifies the project containing the target dataset
+    :param dataset_id: Identifies the dataset to retract deactivated participants from
+    :param sandbox_dataset_id: Identifies the sandbox dataset to store records for dataset_id
+    :returns queries: List of query dictionaries
+    """
+    # gets the deactivated participant dataset to ensure it's up-to-date
+    df = psr.get_deactivated_participants(api_project_id,
+                                          DEACTIVATED_PARTICIPANTS_COLUMNS)
+
+    # To store dataframe in a BQ dataset table named _deactivated_participants
+    destination_table = f'{sandbox_dataset_id}.{DEACTIVATED_PARTICIPANTS}'
+    psr.store_participant_data(df, project_id, destination_table)
+
+    fq_deact_table = f'{project_id}.{destination_table}'
+    deact_table_ref = gbq.TableReference.from_string(f"{fq_deact_table}")
+    # creates sandbox and truncate queries to run for deactivated participant data drops
+    queries = rdp.generate_queries(client, project_id, dataset_id,
+                                   sandbox_dataset_id, deact_table_ref)
+    return queries
+
+
+class RemoveParticipantDataPastDeactivationDate(BaseCleaningRule):
+    """
+    Ensures there is no data past the deactivation date for deactivated participants.
+    """
+
+    def __init__(self, project_id, dataset_id, sandbox_dataset_id,
+                 api_project_id):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect this SQL,
+        append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = (
+            'Sandbox and drop records dated after the date of deactivation for participants'
+            'who have deactivated from the Program.')
+        super().__init__(issue_numbers=['1791'],
+                         description=desc,
+                         affected_datasets=[cdr_consts.COMBINED],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         affected_tables=common.CDM_TABLES)
+        self.api_project_id = api_project_id
+
+    def get_query_specs(self):
+        """
+        This function generates a list of query dicts for ensuring the dates and datetimes are consistent
+
+        :return: a list of query dicts for ensuring the dates and datetimes are consistent
+        """
+
+        deactivation_queries = remove_ehr_data_queries(
+            bq.get_client(self.project_id), self.api_project_id,
+            self.project_id, self.dataset_id, self.sandbox_dataset_id)
```

Comment:
I tried to test the list queries function but it failed. Is there a way to see the queries?

Reply:
It may be because of the participant API. If you comment out those two lines it should work(if you don't want to issue Participant summary API).